src/http_rest_api_handler.cpp

```diff
@@ -441,70 +441,110 @@ Status HttpRestApiHandler::dispatchToProcessor(
+            // Set json parser in invalid state in order to get HasParseError to respond with true
+            doc->Parse("error");
+        } else if (isApplicationJson) {
+            {
+                OVMS_PROFILE_SCOPE("rapidjson parse");
+                doc->Parse(request_body.c_str());
+            }
+            OVMS_PROFILE_SCOPE("rapidjson validate");
+            if (doc->HasParseError()) {
+                return Status(StatusCode::JSON_INVALID, "Cannot parse JSON body");
+            }
 
-        if (!modelNameIt->value.IsString()) {
-            return Status(StatusCode::JSON_INVALID, "model field is not a string");
-        }
+            if (!doc->IsObject()) {
+                return Status(StatusCode::JSON_INVALID, "JSON body must be an object");
+            }
+
+            auto modelNameIt = doc->FindMember("model");
+            if (modelNameIt == doc->MemberEnd()) {
+                return Status(StatusCode::JSON_INVALID, "model field is missing in JSON body");
+            }
 
-        const std::string model_name = modelNameIt->value.GetString();
+            if (!modelNameIt->value.IsString()) {
+                return Status(StatusCode::JSON_INVALID, "model field is not a string");
+            }
 
-        bool isTextGenerationEndpoint = uri.find("completions") != std::string_view::npos;
-        if (isTextGenerationEndpoint) {
-            auto streamIt = doc->FindMember("stream");
-            if (streamIt != doc->MemberEnd()) {
-                if (!streamIt->value.IsBool()) {
-                    return Status(StatusCode::JSON_INVALID, "stream field is not a boolean");
+            bool isTextGenerationEndpoint = uri.find("completions") != std::string_view::npos;
+            if (isTextGenerationEndpoint) {
+                auto streamIt = doc->FindMember("stream");
+                if (streamIt != doc->MemberEnd()) {
+                    if (!streamIt->value.IsBool()) {
+                        return Status(StatusCode::JSON_INVALID, "stream field is not a boolean");
+                    }
+                    streamFieldVal = streamIt->value.GetBool();
                 }
-                streamFieldVal = streamIt->value.GetBool();
+            }
+
+            model_name = modelNameIt->value.GetString();
+            if (model_name.empty()) {
+                isDefault = true;
+            } else {
+                SPDLOG_DEBUG("Model name from deduced from JSON: {}", model_name);
             }
         }
 
+        // Deduce Graph Name from URI since there is no info in JSON or MultiPart
+        if (isDefault) {
+            if (uri.size() <= 4) {  // nothing after "/v3/..."
+                return StatusCode::REST_INVALID_URL;
+            }
+            model_name = std::string(uri.substr(4));
+            SPDLOG_DEBUG("Model name from deduced from URI: {}", model_name);
+            // Set json parser in invalid state in order to get HasParseError to respond with true
+            doc->Parse("error");
```

Comment:
Make it more explicit what happens here and why is it required

ensureDocInErrorState?

Reply:
done